src/vm/analysis/type_checker/natives/mod.rs

```diff
@@ -521,6 +521,27 @@ fn check_get_block_info(
     Ok(TypeSignature::new_option(block_info_prop.type_result())?)
 }
 
+fn check_get_burn_block_info(
+    checker: &mut TypeChecker,
+    args: &[SymbolicExpression],
+    context: &TypingContext,
+) -> TypeResult {
+    check_arguments_at_least(2, args)?;
+
+    let block_info_prop_str = args[0]
+        .match_atom()
+        .ok_or(CheckError::new(CheckErrors::GetBlockInfoExpectPropertyName))?;
```

Comment:
Up to you how you decide to do this, but do you want a separate `CheckErrors` error for burn block properties not being known to the system?